frontend/src/core/hooks/tools/extractPages/useExtractPagesOperation.ts

```diff
@@ -0,0 +1,59 @@
+import apiClient from '@app/services/apiClient';
+import { useTranslation } from 'react-i18next';
+import { ToolType, useToolOperation } from '@app/hooks/tools/shared/useToolOperation';
+import { createStandardErrorHandler } from '@app/utils/toolErrorHandler';
+import { ExtractPagesParameters, defaultParameters } from '@app/hooks/tools/extractPages/useExtractPagesParameters';
+import { pdfWorkerManager } from '@app/services/pdfWorkerManager';
+import { parseSelection } from '@app/utils/bulkselection/parseSelection';
+
+// Convert advanced page selection expression into CSV of explicit one-based page numbers
+async function resolveSelectionToCsv(expression: string, file: File): Promise<string> {
+  // Load PDF to determine max pages
+  const arrayBuffer = await file.arrayBuffer();
+  const pdf = await pdfWorkerManager.createDocument(arrayBuffer, { disableAutoFetch: true, disableStream: true });
+  try {
+    const maxPages = (pdf as any).numPages as number;
```

Comment:
```suggestion
    const maxPages = pdf.numPages as number;
```

I can see another use of `await pdfWorkerManager.createDocument` in the code (`pageCommands.ts`) and that can use `numPages` so I think the cast is probably unnecessary